pkg/apis/smith/v1/types.go

```diff
@@ -183,6 +206,16 @@ func (bs *BundleStatus) ShortString() string {
 	return buf.String()
 }
 
+func (bs *BundleStatus) GetResourceStatus(resName ResourceName) (int, *ResourceStatus) {
+	for i := range bs.ResourceStatus {
+		resStatus := &bs.ResourceStatus[i]
```

Comment:
ok, so I assume you're doing this to extract a pointer to a slice element. But isn't this a bit weird, in that now you're assuming that slice will never be GC'd/mutated? (confused about golang, again)

Reply:
(also, repeating what I said earlier about pointer returns being nasty :P)

Reply:
So, as discussed, there is no obvious much better way to do this. I'll keep it a pointer for now. I've also changed the method on the Bundle to behave the same way.